### PR TITLE
A10M code / ui improvements and fixes

### PR DESCRIPTION
**Pull request by thinkyhead**
Various fixes and improvements to the A10M firmware — mainly the user interface.

[Concise Diff](6/files?w=1)

- Fix a bunch of code formatting and code style issues
- Fix file endings and broken UTF-8 encoding
- Restore axes homed testing to `G26`
- Enable several useful features
- Adjust LCD settings to make the interface more usable
- Increase homing speeds, enable `QUICK_HOME`, add a small raise on homing
- ⚠️ Enable THERMAL PROTECTION for end-user safety. ⚠️ 
- Enable Filament Change (still needs usability improvements)
- Add proper pin definitions for `CONTINUITY_PIN`, `FIL_RUNOUT_PIN`s
- Use fastio `SET_INPUT` and `READ` macros for performance.
- Add G-code `M166` to set the Gradient on the fly.
- Implement `GRADIENT_MIX` as a proper sub-feature.
- Improve the mixer sub-menu and fix several bugs:
  - Rename the "template" submenu to "Gradient"
  - Fix mislabeled "back" items (should be labeled with the parent menu)
  - Fix: Gradient "min/max" tied together when editing
  - Combine mix editing so it applies to both nozzles at once
  - Use the encoder multiplier for faster editing of Z start/end
  - Display all mix values as two reciprocal percentages
  - Add a "Next Gradient" item to cycle through the 4 most common gradients.
  - Add a "Toggle Mix" item to quickly switch extruders.
  - Fix bug: "extra clicks needed to go back" (by using `lcd_goto_previous_menu()` on edit screens)

Please test, compare, and enjoy! More improvements will be coming soon.

---

### CODE REVIEW

File: A10M_marlin1.1.8/Marlin/Configuration.h [hunk 1]
```diff
@@ -144,7 +144,7 @@
 //#define SINGLENOZZLE
 
 /**
- * Pru�a MK2 Single Nozzle Multi-Material Multiplexer, and variants.
```
Comment (Geeetech3D) [inline A10M_marlin1.1.8/Marlin/Configuration.h]: 111

Comment (Geeetech3D) [inline A10M_marlin1.1.8/Marlin/Configuration.h]: 222

**Comment by thinkyhead** [comment]
I've applied some additional improvements just now. I hope you find this useful.

**Comment by thinkyhead** [comment]
Note that the `M166` parameters are now:
#### `M166 S[0|1] A[startz] Z[endz] P[start%] Q[end%]`

**Pull request merged**